### PR TITLE
Logging doc

### PR DESCRIPTION
**Pull request by sgugger**
Adds documentation for the new centralized logger.

### CODE REVIEW

**Comment by stas00** [comment]
How do we get `transformers.logging.*` There is either `transformers.utils.logging.*` or `logging.*` if the latter was imported.

Unrelated, also has the default just changed from INFO to WARN? I rebased my copy and noticed this change.

**Comment by sgugger** [comment]
You get `transformerts.logging.*` after doing `import transformers`. logging is imported in the project init, so there is no need to add the .utils.

**Comment by stas00** [comment]
Ah, I see - the test I was working on was doing `from transformers import logging`. If we follow this in docs it leads to a shorter:

`set_verbosity(logging.INFO)`

and it matches the actual `logging.INFO` from the logging package.

.... but then `from transformers import logging` makes it hard to do `import logging`... same `logging` name. So then:

```
import transformers
transformers.logging.set_verbosity(transformers.logging.INFO)`
```
while being quite verbose, has no collision with the normal `logging` package

Thank you for expanding the docs, @sgugger - this is awesome!

**Comment by sgugger** [comment]
Note that you have the shortcut
```
transformers.logging.set_verbosity_login()
```
but yes, importing logging directly will create a conflict with the logging module.

**Comment by stas00** [comment]
You meant `transformers.logging.set_verbosity_{info|warning|...}` (must be a typo in `login` :)

**Comment by sgugger** [comment]
Oops, fixed my comment.

File: docs/source/main_classes/logging.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+:func:`transformers.logging.get_verbosity` to get the current level of verbosity in the logger and
+:func:`transformers.logging.set_verbosity` to set the verbosity to the level of your choice. In order (from the least
+verbose to the most verbose), those levels (with their corresponding int values in parenthesis) are:
+
+- :obj:`transformers.logging.CRITICAL` or :obj:`transformers.logging.FATAL` (int value, 50): only report the most
+  critical errors.
+- :obj:`transformers.logging.ERROR` (int value, 40): only report errors.
+- :obj:`transformers.logging.WARNING` or :obj:`transformers.logging.WARN` (int value, 30): only reports error and
+  warnings. This the default level used by the library.
+- :obj:`transformers.logging.INFO` (int value, 20): reports error, warnings and basic informations.
```
Comment (LysandreJik) [inline docs/source/main_classes/logging.rst:23]:
```suggestion
- :obj:`transformers.logging.INFO` (int value, 20): reports error, warnings and basic information.
```

File: src/transformers/utils/logging.py [hunk 2]
```diff
@@ -73,8 +73,10 @@ def _reset_library_root_logger() -> None:
 
 
 def get_logger(name: Optional[str] = None) -> logging.Logger:
-    """Return a logger with the specified name.
-    This function is not supposed to be directly accessed by library users.
+    """
+    eturn a logger with the specified name.
```
Comment (LysandreJik) [inline src/transformers/utils/logging.py:77]:
```suggestion
    Return a logger with the specified name.
```

**Review by LysandreJik — APPROVED** [review]
Great, thanks!

**Pull request merged**